src/main/scala/com/fulcrumgenomics/umi/DuplexConsensusCaller.scala

```diff
@@ -0,0 +1,255 @@
+
+        // Call the duplex reads
+        val duplexR1 = duplexConsensus(abR1Consensus, baR2Consensus)
+        val duplexR2 = duplexConsensus(abR2Consensus, baR1Consensus)
+
+        // Convert to SAMRecords and return
+        (duplexR1, duplexR2) match {
+          case (Some(r1), Some(r2)) =>
+            Seq(createSamRecord(r1, FirstOfPair), createSamRecord(r2, SecondOfPair))
+          case _                    =>
+            rejectRecords(recs)
+            Nil
+        }
+      case gs =>
+        unreachable("SAMRecords supplied with more than two distinct MI values.")
+    }
+  }
+
+  /**
+    * Constructs a duplex consensus read from a pair of single strand consensus reads.
+    * If either of the incoming reads are undefined, the duplex read will be undefined.
+    *
+    * @param ab the single-strand consensus from one strand
+    * @param ba the single-strand consensus from the other strand
+    * @return a duplex consensus if one can be built
+    */
+  private[umi] def duplexConsensus(ab: Option[VanillaConsensusRead], ba: Option[VanillaConsensusRead]): Option[DuplexConsensusRead] = {
+    (ab, ba) match {
+      case (Some(a), Some(b)) =>
+        val len = min(a.length, b.length)
+        val id = a.id.substring(0, a.id.lastIndexOf('/')) // Remove the trailing /A or /B that the SS consensuses have
+        val bases = new Array[Byte](len)
+        val quals = new Array[Byte](len)
+
+        forloop(from=0, until=len) { i =>
+          val aBase = a.bases(i)
+          val bBase = b.bases(i)
+          val aQual = a.quals(i)
+          val bQual = b.quals(i)
+
+          val (base, qual) = {
+            if      (aBase == NoCall || bBase == NoCall) (NoCall, PhredScore.MinValue)
+            else if (aBase == bBase) (aBase, (aQual + bQual).toByte)
+            else if (aQual > bQual)  (aBase, (aQual - bQual).toByte)
+            else if (bQual > aQual)  (bBase, (bQual - aQual).toByte)
+            else (NoCall, PhredScore.MinValue)
+          }
+
+          bases(i) = base
+          quals(i) = qual
+        }
+
+        Some(DuplexConsensusRead(id=id, bases, quals, a.truncate(bases.length), b.truncate(bases.length)))
+      case _ =>
+        None
+    }
+  }
+
+  /**
+    * Creates a SAMRecord with a ton of additional tags annotating the duplex read.
+    */
+  override protected def createSamRecord(read: DuplexConsensusRead, readType: ReadType): SAMRecord = {
+    val r = super.createSamRecord(read, readType)
+    val ab = read.abConsensus
+    val ba = read.baConsensus
+
+    // Calculate the total depths and errors across both SS consensus reads
+    val totalDepths = read.abConsensus.depths.zip(read.baConsensus.depths).map(x => x._1 + x._2)
+    val totalErrors = new Array[Short](totalDepths.length)
+    forloop (from=0, until=totalErrors.length) { i =>
+      var e = 0
+      if (read.bases(i) == ab.bases(i)) e += ab.errors(i) else e += ab.depths(i) - ab.errors(i)
```

Comment:
You're not wrong, and I did think about this.  The choice is between making things quite a bit more complicated (the vanilla consensus read would now need to carry around counts of all four bases at each position instead of depths and errors, so that they could be used in duplex calling, not because they are needed for vanilla calling) vs. a minor inaccuracy in an edge case that I think it not terribly important.

I say "not terribly important" because the only way this can arise is if the two single-strand consensus reads disagree at a position _and_ there's > 2 different bases observed in one or both of them.  If the two disagree the consensus call is already going to be << Q45 and the annotated error rate is already going to be high.  

So really it's a question of whether this is worth fixing to be 100% accurate, when I think it's functionally ok.